docs/power.rst

```diff
@@ -0,0 +1,161 @@
+  on battery power.
+  When the board wakes up it will restart and execute your programme from the
+  beginning.
+
+The micro:bit board contains two microcontrollers (MCUs), which can be
+independently awake or asleep:
+
+- **Target MCU** - Where MicroPython and your code run
+- **Interface MCU** - A secondary microcontroller that provides the USB
+  functionality, like the ``MICROBIT`` USB drive, and the USB serial interface
+
+.. image:: power-mcus.png
+
+Each microcontroller can go into its individual "MCU power mode", and the
+combination of these modes defines the "board power mode".
+The Python code can request a "board power mode", which will
+set the Target MCU into specific state, but the Interface "MCU power mode"
+will depend on the micro:bit power source, i.e. if it's powered via USB or
+battery.
+
++------------------+---------------+--------------------+
+| .. centered:: USB Powered (Interface always awake)    |
++------------------+---------------+--------------------+
+| Board Power mode | Target mode   |  Interface mode    |
++==================+===============+====================+
+| **Deep Sleep**   | 💤 Sleep      | ⏰ Awake           |
++------------------+---------------+--------------------+
+| **Off**          | 📴 Power Down | ⏰ Awake           |
+|                  |               | (red LED blinking) |
++------------------+---------------+--------------------+
+
++------------------+---------------+------------------+
+| .. centered:: Battery Powered                       |
++------------------+---------------+------------------+
+| Board Power mode | Target mode   |  Interface mode  |
++==================+===============+==================+
+| **Deep Sleep**   | 💤 Sleep      | 💤 Sleep         |
++------------------+---------------+------------------+
+| **Off**          | 📴 Power Down | 📴 Power Down    |
++------------------+---------------+------------------+
+
+We can see in these tables that there are two "MCU low power modes":
+
+- Sleep: A low power mode where the MCU can be woken up and continue operation.
+- Power Down: The lowest power mode for an individual MCU, where waking up
+  reset operation.
+
+Functions
+=========
+
+.. py:function:: off()
+
+    Power down the board to the lowest possible power mode.
+
+    This is the equivalent to pressing the reset button for a few second,
+    to set the board in "Off mode".
+
+    The Target MCU will go into Power Down mode. The Interface MCU will go to
+    Power Down on battery or stay awake on USB, where it will blink the red
+    LED near the USB connector.
+
+    The micro:bit will only wake if the reset button is pressed, or if on
+    battery power, when a USB cable is connected.
+
+    The the board wakes up it will start for a reset state, so your programme
+    will start running from the beginning.
+
+
+.. py:function:: deep_sleep(ms=None, pins=None, buttons=None, run_every=False)
```

Comment:
`pins` and `buttons` could actually be combined into, eg, `events`.  Because the code can check the type of the objects passed in.  Eg `deep_sleep(events=(button_a, pin0))`.

Reply:
Yeah, having the buttons for only 2 options doesn't feel quite right. Although I'm not 100% sure about the word `events` though, it doesn't feel very obvious `deep_sleep(events=(pin1,pin2))`, but it's warming up on me.

Do you think using `pins` and include buttons in the same argument would be confusing? `deep_sleep(pins=(pin1, button_a))`

Reply:
How about `wake_on` or `wake_by`?  Eg `deep_sleep(wake_on=(pin1, button_a))`.

Reply:
`wake_on` sounds good 👍 
Added in 082574457fd3ec725e50730132f69bddbf279827.

Reply:
OK, now implemented.